sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -604,12 +604,11 @@ public virtual async Task StartProcessingAsync(CancellationToken cancellationTok
         public virtual async Task StopProcessingAsync(CancellationToken cancellationToken = default)
         {
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            await RunningTaskSemaphore.WaitAsync().ConfigureAwait(false);
 
-            if (ActiveLoadBalancingTask != null)
+            try
             {
-                await RunningTaskSemaphore.WaitAsync().ConfigureAwait(false);
-
-                try
+                if (ActiveLoadBalancingTask != null)
                 {
                     cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
```

Comment:
I don't get why you moved these around.

https://github.com/Azure/azure-sdk-for-net/blob/217f1432471e8ff98bce9fc99d18779e7625b59a/sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs#L609-L616

It feels like there's no point in the double check anymore.

Reply:
We had a race condition that my tests hit.    We could await the start,  trigger an exception,  request stop, and then read `IsRunning` as true because the null check caused stop to bail out early.

Reply:
You're right.   I missed that double-check in there.   I'll remove in the samples PR.  There's no real impact to it being there for this release.